src/Workspaces/Core/Desktop/Workspace/FileTextLoader.cs

```diff
@@ -72,7 +72,6 @@ protected virtual SourceText CreateText(Stream stream, Workspace workspace)
             TextAndVersion textAndVersion;
             using (var stream = FileUtilities.OpenAsyncRead(_path))
             {
```

Comment:
Does this go through your async stream? If so, you're going to be basically bypassing overlapped I/O because you don't delegate ReadAsync/WriteAsync down to the wrapped stream.